framework/src/engine/generator/generator.ts

```diff
@@ -622,10 +625,13 @@ export class Generator {
 	}
 
 	private _handleFinalizedHeightChanged(from: number, to: number): Promise<void>[] {
+		if (from >= to) {
+			return [];
+		}
 		const promises = [];
 		const stateStore = new StateStore(this._blockchainDB);
 		for (const [address, pairs] of this._keypairs.entries()) {
-			for (let height = from + 1; height <= to; height += 1) {
+			for (let height = from + 1; height < to; height += 1) {
 				promises.push(
```

Comment:
In this case, we should first check `existBFTParameters(height + 1) == true` and otherwise not create single commits.

Reply:
that is checked in the function `_certifySingleCommitForChangedHeight`

Reply:
True, I got confused with the functions again. All good.